src/main/java/org/spongepowered/api/command/conversation/QuestionResult.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.command.conversation;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Optional;
+
+/**
+ * What to return from your answer handler on a question, to allow moving on to
+ * a next question, the same question, or to end the conversation.
+ */
+public class QuestionResult {
+
+    /**
+     * Gets a question result which will end the question.
+     *
+     * @return A question result which ends the question
+     */
+    public static QuestionResult end() {
+        return new QuestionResult(QuestionResultType.END);
```

Comment:
![image](https://cloud.githubusercontent.com/assets/18372958/26374791/21615df8-3fcc-11e7-9546-35fa8149f1bc.png)
in the `SpongeConversation` class.

It's essentially to include the type in the cause, but I have had doubts on if that is necessary. Maybe I should just put the ConversationEndType in the cause, I'm not sure.